data-pipeline/src/pipeline/database.py

```diff
@@ -29,15 +33,149 @@
     query={"driver": "ODBC Driver 18 for SQL Server"} | args,
 )
 
-engine = create_engine(connection_url)
+engine = create_engine(connection_url, fast_executemany=True)
 
 
-@event.listens_for(engine, "before_cursor_execute")
-def receive_before_cursor_execute(
-    conn, cursor, statement, params, context, executemany
-):
-    if executemany:
-        cursor.fast_executemany = True
+@dataclass
+class TempTable:
+    name: str
+    target: str
+    columns: list[str]
+
+    def __str__(self):
+        return self.name
+
+
+def _get_table_columns(table: str) -> list[str]:
+    """
+    Retrieve the columns for a given table.
+
+    The source-of-truth for the table formats should remain the DB
+    itself: here, we retrieve the list of expected columns directly
+    from the table's schema.
+
+    :param table: DB table for which to retrieve columns
+    :return: column names
+    """
+    sql = textwrap.dedent("""
```

Comment:
Generally, these changes lean towards deriving as much from the DB as possible: here, for instance, the temp. table is derived from the target table, not defined in the code here.